ModelVale/ModelVale/CustomComponents/HealthBarView.m

```diff
@@ -118,111 +100,30 @@ - (UIBezierPath*) getBarPath: (CGFloat)endRoundness withWidthPercent: (CGFloat)w
-    pathAnimation.fromValue = (__bridge id)[startPath CGPath];
-//    pathAnimation.toValue = (__bridge id)[filledBarPath CGPath];
-    pathAnimation.duration = animationDuration;
-    [self.healthShapeLayer addAnimation:pathAnimation forKey:@"archBar"];
-    [self.barShapeLayer addAnimation:pathAnimation forKey:@"archBar"];
-}
-
-- (void) fillProgressBar: (UIBezierPath*) filledBarPath {
-
-    UIBezierPath* startPath = [self getBarPath:10 withWidthPercent:0];
+- (void) animateFillingHealthBar: (UIBezierPath*)filledBarPath layer: (CALayer*)layer {
+    
+    UIBezierPath* startPath = [self getBarPath:0];
     CABasicAnimation * pathAnimation = [CABasicAnimation animationWithKeyPath:@"path"];
     pathAnimation.fromValue = (__bridge id)[startPath CGPath];
     pathAnimation.toValue = (__bridge id)[filledBarPath CGPath];
-    pathAnimation.duration = animationDuration;
-    [self.healthShapeLayer addAnimation:pathAnimation forKey:@"fillBar"];
+    pathAnimation.duration = self.animationDuration;
+    [layer addAnimation:pathAnimation forKey:@"fillBar"];
 }
 
-//MARK: XP Animations
-
-- (UIImageView*) getXPImageView: (CGFloat) size label: (NSString*) label {
-    UIImageView* xpImView = [UIImageView new];
-    xpImView.frame = CGRectMake(0, 0, size, size);
-    xpImView.image = [UIImage imageNamed:label];
-    return xpImView;
-}
-
-- (CAShapeLayer*) getXPBubbleLayer: (CGPoint) XPStart {
-    CAShapeLayer* XPLayer = [CAShapeLayer new];
-    XPLayer.fillColor = [UIColor colorWithWhite:1 alpha:0].CGColor; // transparent
-    return XPLayer;
-}
-
-- (UIBezierPath*) getXPPath: (CGPoint)XPStart XPEnd: (CGPoint) XPEnd {
-    //ASSUMPTION Moving right to left and up
-    UIBezierPath* path = [UIBezierPath new];
-    [path moveToPoint:XPStart];
-    CGPoint controlOne = CGPointMake(XPEnd.x - 0.5*(XPStart.x - XPEnd.x), XPStart.y - 50);
-    CGPoint controlTwo = CGPointMake(XPStart.x + 0.9*(XPStart.x - XPEnd.x), XPStart.y - 0.5*(XPStart.y - XPEnd.y));
-    [path addCurveToPoint:XPEnd controlPoint1:controlOne controlPoint2:controlTwo];
-    return path;
-}
-
-- (void) animateXPPath: (CALayer*) XPLayer path: (UIBezierPath*) XPPath {
-    CAKeyframeAnimation * pathAnimation = [CAKeyframeAnimation animationWithKeyPath:@"position"];
-    pathAnimation.path = XPPath.CGPath;
-    pathAnimation.duration = animationDuration;
-    pathAnimation.fillMode = kCAFillModeForwards;
-    pathAnimation.removedOnCompletion = NO;
-    [XPLayer addAnimation:pathAnimation forKey:@"flyingXP"];
+- (void) addGradientToHealthBar: (CAShapeLayer*)healthBarLayer gradWidth: (NSInteger)gradWidth {
+    
+    CAGradientLayer* gradient = [CAGradientLayer new];
+    CGPoint start = CGPointMake(0, 0);
+    CGPoint end = CGPointMake(1, 0);
+    gradient.startPoint = start;
+    gradient.endPoint = end;
+    gradient.frame = self.bounds;
+    UIColor* violet = [UIColor colorWithRed:125.0f/255.0f green:65.0f/255.0f blue:205.0f/255.0f alpha:1.0f];
```

Comment:
I wasn't exactly sure how to go about this since doing
`
UIColor* const violet = [UIColor colorWithRed:125.0f/255.0f green:65.0f/255.0f blue:205.0f/255.0f alpha:1.0f];

`
Can't be stored as a constant since it needs the UIColor class, I think. I could store an array of the rgba values instead?